code/datums/components/plumbing/_plumbing.dm

```diff
@@ -42,25 +42,28 @@
 	reagents = AM.reagents
 	turn_connects = _turn_connects
 
-	if(custom_receiver)
-		recipient_reagents_holder = custom_receiver
-	else
-		recipient_reagents_holder = AM.reagents
+	set_recipient_reagents_holder(custom_receiver ? custom_receiver : AM.reagents)
 
+	if(start)
+		//timer 0 so it can finish returning initialize, after which we're added to the parent.
+		//Only then can we tell the duct next to us they can connect, because only then is the component really added. this was a fun one
+		addtimer(CALLBACK(src, .proc/enable), 0)
```

Comment:
This looks like something that should be systemic or you could register to component added signal instead and unregister in response.

Reply:
Yeah sleeps, even for 0, to deal with init issues isn't the way to go. React to something else when completed.

Reply:
Do you mean just send a signal?

Reply:
Yeah, you could do what antur mentioned or listen for some other signal that happens around when you want it. By react though I just triggering this code by some method when other code runs.